tcl.html.markdown

```diff
@@ -0,0 +1,377 @@
+---
+language: Tcl
+contributors:
+    - ["Poor Yorick", "http://pooryorick.com/"]
+---
+
+Tcl was created by [John Ousterhout](http://wiki.tcl.tk/John Ousterout) as a
+reusable scripting language for chip design tools he was creating.  In 1997 he
+was awarded the [ACM Software System
+Award](http://en.wikipedia.org/wiki/ACM_Software_System_Award) for Tcl.  Tcl
+can be used both as an embeddable scripting language and as a general
+programming language.  It can also be used as a portable C library, even in
+cases where no scripting capability is needed, as it provides data structures
+such as dynamic strings, lists, and hash tables.  The C library also provides
+portable functionality for loading dynamic libraries, string formatting and
+code conversion, filesystem operations, network operations, and more.
+
+Tcl is a pleasure to program in.  Its discipline of exposing all programmatic
+functionality as commands, including things like loops and mathematical
+operations that are usually baked into the syntax of other languages, allows it
+to fade into the background of whatever domain-specific functionality a project
+needs.  Its design of exposing all values as strings, while internally caching
+a structured representation, bridges the world of scripting and systems
+programming in the best way.  Even Lisp is more syntactically heavy than Tcl.
+
+
+
+```tcl
+#! /bin/env tclsh
+
+################################################################################
+## 1. Guidelines 
+################################################################################
+
+# Tcl is not Bash or C!  This needs to be said because standard shell quoting
+# habits almost work in Tcl and it is common for people to pick up Tcl and try
+# to get by with syntax they know from another language.  It works at first,
+# but soon leads to frustration with more complex scripts.
+ 
+# Braces are just a quoting mechanism, not a code block constructor or a list
+# constructor.  Tcl doesn't have either of those things.  Braces are used,
+# though, to escape special characters in procedure bodies and in strings that
+# are formatted as lists.
+
+
+################################################################################
+## 2. Syntax 
+################################################################################
+
+# Every line is a command.  The first word is the name of the command, and
+# subsequent words are arguments to the command. Words are delimited by
+# whitespace.  Since every word is a string, no escaping is necessary in the
+# simple case.
+
+set greeting1 Sal 
+set greeting2 ut
+set greeting3 ations
+
+
+#semicolon also delimits commands
+
+set greeting1 Sal; set greeting2 ut; set greeting3 ations 
+
+
+# Dollar sign introduces variable substitution
+
+set greeting $greeting1$greeting2
+
+
+# Bracket introduces command substitution
+
+set greeting $greeting[set greeting3]

```

Comment:
if you havn't done any Tcl before, this is confusing. Maybe add a note explaing that `set` with only one argument returns the value?